.github/workflows/Doxygen.yml

```diff
@@ -46,11 +46,79 @@ jobs:
         with:
           doxyfile-path: './Doxyfile'
           working-directory: ./Doc
-  
-      # Deploys the generated documentation to GitHub Pages
-      - name: Deploy
-        uses: peaceiris/actions-gh-pages@v4
+
+      - name: Upload Doxygen Artifact
+        uses: actions/upload-artifact@v4
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./Doc/html
+          retention-days: 1
+          name: DoxygenHTML
+          path: Doc/html
+          
+  Sphinx:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
     
+    container:
+      image: ghcr.io/mach3-software/mach3:alma9latest
+
+    needs: Doxygen
+      
+    steps:
+    
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v4
+
+      - name: Check Python 
+        run: python3 --version
+
+      - name: Download Doxygen Artifact
+        uses: actions/download-artifact@master
+        with:
+          name: DoxygenHTML
+          path: to-upload/
+
+      ## EM: Now we delete the artifact so it doesn't use up our allowance
+      - uses: geekyeggo/delete-artifact@v5
+        with:
+          name: DoxygenHTML
+          
+      - name: Build MaCh3
+        run: | 
+          mkdir build
+          cd build 
+          cmake -DMaCh3_PYTHON_ENABLED=ON ../
+          make 
```

Comment:
most gitbug action have 4 threads
make -j4

this will reduce github action time we use ;)